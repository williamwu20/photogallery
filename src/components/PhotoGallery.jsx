import React from 'react'
import ReactSwipe from 'react-swipe'
import PropTypes from 'prop-types'

export default class PhotoGallery extends React.Component {
    render() {
        return (
            <div>
                <ReactSwipe
                    className='carousel'
                    swipeOptions={{ continuous: true }}
                    childCount={this.props.photos.length}
                    ref={el => (this.reactSwipeEl = el)}
                >
                    {this.props.photos.map((photo, i) => (
                    <div className='item' key={i}>
                        <img src={photo.url} />
                        <p>{photo.caption}</p>
                    </div>))}
                </ReactSwipe>
                <div className='controls'>
                    <button onClick={() => this.reactSwipeEl.prev()}>Previous</button>
                    <button onClick={() => this.reactSwipeEl.next()}>Next</button>
                </div>
            </div>
        )
    }
}