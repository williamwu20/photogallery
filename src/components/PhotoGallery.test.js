import React from 'react'
import { shallow } from 'enzyme'
import PhotoGallery from './PhotoGallery.jsx'
import '../jest.setup'

describe(PhotoGallery, () => {
  const photos = [{url: 'photo_url', caption: 'photo_caption'}]
  describe('props', () => {
    describe('.photos', () => {
      it('takes in an array of photo objects', () => {
        const wrapper = shallow(<PhotoGallery photos={photos} />)
        expect(wrapper.find('img').prop('src')).toEqual(photos[0].url)
        expect(wrapper.find('p').text()).toEqual(photos[0].caption)
      })
    })
  })

  describe('render', () => {
    it('renders previous and next buttons', () => {
      const wrapper = shallow(<PhotoGallery photos={photos} />)
      expect(wrapper.find('.previous').length).toEqual(1)
      expect(wrapper.find('.next').length).toEqual(1)
    })
  })
})
