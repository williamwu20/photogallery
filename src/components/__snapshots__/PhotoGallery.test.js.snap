// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGallery render matches the snapshot 1`] = `
<div>
  <div
    className="react-swipe-container carousel"
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
        "visibility": "hidden",
      }
    }
  >
    <div
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        className="item"
        style={
          Object {
            "float": "left",
            "position": "relative",
            "transitionProperty": "transform",
            "width": "100%",
          }
        }
      >
        <img
          src="photo_url"
        />
        <p>
          photo_caption
        </p>
      </div>
    </div>
  </div>
  <div
    className="controls"
  >
    <button
      className="previous"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="next"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
